src/plugins/saved_objects_management/server/routes/find.ts

```diff
@@ -98,7 +98,11 @@ export const registerFindRoute = (
         ...req.query,
         fields: undefined,
         searchFields: [...searchFields],
-        workspaces: req.query.workspaces ? Array<string>().concat(req.query.workspaces) : undefined,
+        ...(req.query.workspaces
+          ? {
+              workspaces: Array<string>().concat(req.query.workspaces),
+            }
+          : {}),
```

Comment:
I think we need to filter out `*` here, if we pass workspaces as `*`, it will able to query all workspace.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f52b5d2e9f0f1b93f3f22246783e992820e1213c/src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts#L48-L58

Reply:
We have another PR to fix the * issue: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9409. I think it should be fine for this PR to leave *.